convert_floating argument is missing in convert_dtypes

**Describe the bug**

In both `NDFrame.convert_dtypes()` and `Series.convert_dtypes()`, the annotation for the `convert_floating` argument that was introduced in Pandas 1.2.0 is missing. This causes IDEs and type checkers to report an unexpected argument when using `convert_floating`.

**To Reproduce**

1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
   ```python
   import pandas as pd

   df = pd.DataFrame()
   df = df.convert_dtypes(convert_floating=False)
   ```
2. Indicate which type checker you are using (`mypy` or  `pyright`).
   `mypy`
3. Show the error message received from that type checker while checking your example.
   ```
   temp.py:4: error: Unexpected keyword argument "convert_floating" for "convert_dtypes" of "NDFrame"; did you mean "convert_string"?  [call-arg]
   ```


**Please complete the following information:**
 - OS: Linux
 - OS Version: Ubuntu 20.04.6 LTS
 - python version: 3.10.13
 - version of type checker: 1.5.1
 - version of installed `pandas-stubs`: 1.5.3.230321


**Additional context**
I also tried using `pandas-stubs` 2.1.1.230928 intead and got the same error message with an additional hint:
```
temp.py:4: error: Unexpected keyword argument "convert_floating" for "convert_dtypes" of "NDFrame"; did you mean "convert_string"?  [call-arg]
/home/<user>/.cache/pypoetry/virtualenvs/<project>-py3.10/lib/python3.10/site-packages/pandas-stubs/core/generic.pyi:371: note: "convert_dtypes" of "NDFrame" defined here
```
We are using `pandas-stubs` versioned at `^1.5.2.0` because we have not made the switch to `pandas` 2.0 yet.

--- a/pandas-stubs/core/generic.pyi
+++ b/pandas-stubs/core/generic.pyi
@@ -371,6 +371,7 @@ class NDFrame(indexing.IndexingMixin):
         convert_string: _bool = ...,
         convert_integer: _bool = ...,
         convert_boolean: _bool = ...,
+        convert_floating: _bool = ...,
         dtype_backend: DtypeBackend = ...,
     ) -> Self: ...
     def fillna(

--- a/pandas-stubs/core/series.pyi
+++ b/pandas-stubs/core/series.pyi
@@ -106,7 +106,6 @@ from pandas._typing import (
     ComplexDtypeArg,
     CompressionOptions,
     Dtype,
-    DtypeBackend,
     DtypeObj,
     FilePath,
     FillnaOptions,
@@ -1245,14 +1244,6 @@ class Series(IndexOpsMixin[S1], NDFrame):
     ) -> Series: ...
     def copy(self, deep: _bool = ...) -> Series[S1]: ...
     def infer_objects(self) -> Series[S1]: ...
-    def convert_dtypes(
-        self,
-        infer_objects: _bool = ...,
-        convert_string: _bool = ...,
-        convert_integer: _bool = ...,
-        convert_boolean: _bool = ...,
-        dtype_backend: DtypeBackend = ...,
-    ) -> Series[S1]: ...
     @overload
     def ffill(
         self,

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -2795,6 +2795,12 @@ def test_suffix_prefix_index() -> None:
     )
 
 
+def test_convert_dtypes_convert_floating() -> None:
+    df = pd.DataFrame({"A": [1, 2, 3, 4], "B": [3, 4, 5, 6]})
+    dfn = df.convert_dtypes(convert_floating=False)
+    check(assert_type(dfn, pd.DataFrame), pd.DataFrame)
+
+
 def test_convert_dtypes_dtype_backend() -> None:
     df = pd.DataFrame({"A": [1, 2, 3, 4], "B": [3, 4, 5, 6]})
     dfn = df.convert_dtypes(dtype_backend="numpy_nullable")

--- a/tests/test_series.py
+++ b/tests/test_series.py
@@ -2701,6 +2701,12 @@ def test_prefix_summix_axis() -> None:
         s.add_suffix("_item", axis="columns")  # type: ignore[arg-type] # pyright: ignore[reportGeneralTypeIssues]
 
 
+def test_convert_dtypes_convert_floating() -> None:
+    df = pd.Series([1, 2, 3, 4])
+    dfn = df.convert_dtypes(convert_floating=False)
+    check(assert_type(dfn, "pd.Series[int]"), pd.Series, np.integer)
+
+
 def test_convert_dtypes_dtype_backend() -> None:
     s = pd.Series([1, 2, 3, 4])
     s1 = s.convert_dtypes(dtype_backend="numpy_nullable")

